src/App.tsx

```diff
@@ -1,26 +1,143 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorComponent } from './components/ErrorComponent/ErrorComponent';
+import { Footer } from './components/Footer/Footer';
+import { FilterStatusType } from './types/FilterStatusType';
+import cn from 'classnames';
+import { FormComponent } from './components/FormComponent/FormComponent';
+import { ErrorMessage } from './types/ErrorStatusType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [showError, setShowError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterStatus, setFilterStatus] = useState(FilterStatusType.All);
+  const [isCompletedTodosExist, setIsCompletedTodosExist] = useState(true);
+  const [numberOfNotCompletedTodos, setNumberOfNotCompletedTodos] = useState(0);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [needToRefresh, setNeedToRefresh] = useState(true);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isTempTodoLoading, setIsTempTodoLoading] = useState(false);
+  const [todoIdsToDelete, setTodoIdsToDelete] = useState<number[]>([]);
+  const [isFocusTitleInput, setIsFocusTitleInput] = useState(false);
+
+  useEffect(() => {
+    if (showError) {
+      setTimeout(() => {
+        setShowError(false);
+      }, 3000);
+    }
+  }, [showError]);
+
+  useEffect(() => {
+    if (todos.some(todo => todo.completed)) {
+      setIsCompletedTodosExist(true);
+    } else {
+      setIsCompletedTodosExist(false);
+    }
+
+    let counter = 0;
+
+    for (const todo of todos) {
+      if (!todo.completed) {
+        counter++;
+      }
+    }
+
+    setNumberOfNotCompletedTodos(counter);
+  }, [todos]);
+
+  useEffect(() => {
+    switch (filterStatus) {
+      case FilterStatusType.Active:
+        setVisibleTodos(todos.filter(todo => !todo.completed));
+        break;
+      case FilterStatusType.Completed:
+        setVisibleTodos(todos.filter(todo => todo.completed));
+        break;
+      default:
+        setVisibleTodos(todos);
+    }
+  }, [filterStatus, todos]);
+
+  useEffect(() => {
+    if (needToRefresh) {
```

Comment:
what does it mean?
you should work with state we don't want to make extra calls on server

Reply:
looks like kostyl ;)